metrics-operator/api/v1alpha3/analysis_webhook.go

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha3
+
+import (
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	ctrl "sigs.k8s.io/controller-runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+// Analysislog is for logging in this package.
+var Analysislog = logf.Log.WithName("analysis-webhook")
+
+func (a *Analysis) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(a).
+		Complete()
+}
+
+//+kubebuilder:webhook:path=/validate-metrics-keptn-sh-v1alpha3-analysis,mutating=false,failurePolicy=fail,sideEffects=None,groups=metrics.keptn.sh,resources=analyses,verbs=create;update,versions=v1alpha3,name=analysis.kb.io,admissionReviewVersions=v1
+
+var _ webhook.Validator = &Analysis{}
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
+func (a *Analysis) ValidateCreate() (admission.Warnings, error) {
+	Analysislog.Info("validate create", "name", a.Name)
```

Comment:
Maybe it is worth to add also the namespace 🤔 Most likely different app in diff ns will use the same name and if we have a gitops tool sync all of them it would be handy to know which ns f'd up stuff :)

Reply:
yep, that's a good idea, will add that